src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/OrdinaryMethodReferenceFinder.cs

```diff
@@ -27,9 +28,53 @@ protected override ValueTask<ImmutableArray<ISymbol>> DetermineCascadedSymbolsAs
     {
         // If it's a delegate method, then cascade to the type as well.  These guys are
         // practically equivalent for users.
-        return symbol.ContainingType.TypeKind == TypeKind.Delegate
-            ? new(ImmutableArray.Create<ISymbol>(symbol.ContainingType))
-            : new(GetOtherPartsOfPartial(symbol));
+        if (symbol.ContainingType.TypeKind == TypeKind.Delegate)
+            return new([symbol.ContainingType]);
+
+        using var _ = ArrayBuilder<ISymbol>.GetInstance(out var result);
+
+        result.AddRange(GetOtherPartsOfPartial(symbol));
+        CascadeToExtensionImplementation(symbol, result);
+        CascadeFromExtensionImplementation(symbol, result);
+
+        return new(result.ToImmutableAndClear());
+    }
+
+    private static void CascadeToExtensionImplementation(IMethodSymbol symbol, ArrayBuilder<ISymbol> result)
+    {
+        // If the given symbol is an extension member, cascade to its implementation method
+        if (symbol.AssociatedExtensionImplementation is { } associatedExtensionImplementation)
+            result.Add(associatedExtensionImplementation);
+    }
+
+    private static void CascadeFromExtensionImplementation(IMethodSymbol symbol, ArrayBuilder<ISymbol> result)
+    {
+        // If the given symbol is an implementation method of an extension member, cascade to the extension member itself
+        var containingType = symbol.ContainingType;
+        if (containingType is null || !containingType.MightContainExtensionMethods || !symbol.IsStatic)
+            return;
+
+        var implementationDefinition = symbol.OriginalDefinition;
```

Comment:
i don't think you need to do .OriginalDefinition within the REfFinder code.  Everything is OriginalDefs here.